CIScripts/Tests/TestConfigurationUtils.ps1

```diff
@@ -0,0 +1,174 @@
+class DockerNetworkConfiguration {
+    [string] $TenantName;
+    [string] $NetworkName;
+}
+
+class DockerDriverConfiguration {
+    [string] $Username;
+    [string] $Password;
+    [string] $AuthUrl;
+    [string] $ControllerIP;
+    [DockerNetworkConfiguration] $NetworkConfiguration;
+}
+
+class TestConfiguration {
+    [DockerDriverConfiguration] $DockerDriverConfiguration;
+    [string] $AdapterName;
+    [string] $VMSwitchName;
+    [string] $ForwardingExtensionName;
+}
+
+function Enable-VRouterExtension {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [string] $AdapterName,
+           [Parameter(Mandatory = $true)] [string] $VMSwitchName,
+           [Parameter(Mandatory = $true)] [string] $ForwardingExtensionName)
+
+    Write-Host "Enabling Extension"
+
+    Invoke-Command -Session $sess -ScriptBlock {
+        New-ContainerNetwork -Mode Transparent -NetworkAdapterName $Using:AdapterName -Name test_network
+        Enable-VMSwitchExtension -VMSwitchName $Using:VMSwitchName -Name $Using:ForwardingExtensionName | Out-Null
+    }
+}
+
+function Disable-VRouterExtension {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [string] $VMSwitchName,
+           [Parameter(Mandatory = $true)] [string] $ForwardingExtensionName)
+
+    Write-Host "Disabling Extension"
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        Disable-VMSwitchExtension -VMSwitchName $Using:VMSwitchName -Name $Using:ForwardingExtensionName -ErrorAction SilentlyContinue | Out-Null
+        Get-ContainerNetwork | Remove-ContainerNetwork -Force
```

Comment:
I think it would be better if we filter container networks for adapter name, before removing everything.